File: examples/node/gemini_example.ts
/**
 * Build a Gemini generateContent payload for the api-designer skill.
 *
 * Run with:
 *   npx tsx examples/node/gemini_example.ts
 */
import path from "node:path";
import { fileURLToPath } from "node:url";
import { load, getAdapter } from "../../packages/usf-ts/src/index.js";

const here = path.dirname(fileURLToPath(import.meta.url));
const skillPath = path.resolve(here, "../../skills/api-designer.md");

async function main() {
  const skill = await load(skillPath);
  const compiled = await skill.buildPrompt({
    feature: "A notification service that fans out events to email, SMS, and push.",
    style: "REST",
  });
  const payload = getAdapter("gemini").render(compiled);
  console.log(JSON.stringify(payload, null, 2));
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: examples/python/gemini_example.py
"""Run the api-designer skill against Google Gemini.

Prereq:
    pip install -e packages/usf-py[gemini]
    export GEMINI_API_KEY=...

Run:
    python examples/python/gemini_example.py
"""
from __future__ import annotations

from pathlib import Path

from usf import load

SKILL = Path(__file__).resolve().parents[2] / "skills" / "api-designer.md"


def main() -> None:
    skill = load(SKILL)
    output = skill.run(
        "gemini",
        {
            "feature": "A notification service that fans out events to email, SMS, and push.",
            "style": "REST",
        },
    )
    print(output)


if __name__ == "__main__":
    main()
